cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -192,22 +193,18 @@ MapBuilderBridge::GetTrajectoryStates() {
 
 visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodeList() {
   visualization_msgs::MarkerArray trajectory_node_list;
-  const auto all_trajectory_nodes =
-      map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
-  for (int trajectory_id = 0;
-       trajectory_id < static_cast<int>(all_trajectory_nodes.size());
-       ++trajectory_id) {
-    const auto& single_trajectory_nodes = all_trajectory_nodes[trajectory_id];
+  const auto nodes = map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
+  for (const int trajectory_id : nodes.trajectory_ids()) {
     visualization_msgs::Marker marker =
         CreateTrajectoryMarker(trajectory_id, node_options_.map_frame);
 
-    for (const auto& node : single_trajectory_nodes) {
-      if (node.constant_data == nullptr) {
+    for (const auto& node_id_data : nodes.trajectory(trajectory_id)) {
+      if (node_id_data.data.constant_data == nullptr) {
```

Comment:
This isn't possible anymore, right? Maybe fix in your follow up PR to get this merged quicker.

Reply:
roger roger